app/forms/concerns/wizard.rb

```diff
@@ -0,0 +1,61 @@
+module Wizard
+  extend ActiveSupport::Concern
+
+  included do
+    attr_writer :state_store
+    attr_accessor :current_step, :review_mode
+  end
+
+  def initialize(state_store, params = {})
+    @state_store = state_store
+    sanitize_attributes(params)
+
+    super(last_saved_state.deep_merge(params))
+
+    enter_review_mode! if current_step.eql?('check')
+  end
+
+  def next_step
+    :check
+  end
+
+  def previous_step
```

Comment:
It would be really cool if we could commonize the logic around review_mode and previous/next steps in this module. Ideally, no-one else needs to know about review_mode. Something like:
```
def previous_step
  if review_mode?
    :check 
  else 
    get the step from the wizard
  end
end
```

Reply:
There is no common pattern between the different wizards (besides that they tend to go to :check if their conditions are not satisfied, and even that doesn't apply to all of them) so I think trying to achieve this right now will be overengineering.

Reply:
I think there should be a common pattern between wizards that use a review mode. My understanding was that the existing checking answers flag is used when a user presses a change link from the check page, they go back to the relevant step. Then once they click continue, they jump right back to the check page. It seems like it would be useful to have that logic in one place, even if it were an optional thing for wizards with a check page. I might have misunderstood what review_mode means though?

If not, does that suggest that the logic around entering review mode doesn't belong in this module either?